File: 01-1.Two_Sum(E)/Hashmap.js
// Approach Hashmap - T O(N), S O(N)

var twoSum = function (nums, target) {
  const valIndexMap = {} // value : index

  for (let i = 0; i < nums.length; i++) {
    diff = target - nums[i]

    if (diff in valIndexMap) {
      return [i, valIndexMap[diff]]
    }

    valIndexMap[nums[i]] = i
  }
}

const nums = [2, 3, 11, 15]
const target = 14

const result = twoSum(nums, target)
console.log(result)

File: 01-1.Two_Sum(E)/Hashmap.py
def two_sum(nums, target):
    cache = {}  # {number : index}
    for i, num in enumerate(nums):
        key = target - num
        if key in cache:
            return cache[key], i
        cache[num] = i


print(two_sum([2, 7, 11, 15], 9))
